src/sagas/disks.js

```diff
@@ -1,21 +1,54 @@
 import { takeEvery, put, select } from 'redux-saga/effects'
 
-import { CREATE_DISK_FOR_VM, REMOVE_DISK, EDIT_VM_DISK } from '../constants'
-import Api from '../ovirtapi'
-import { callExternalAction, delay, delayInMsSteps } from './utils'
-import { fetchDisks } from './index'
+import Api, { Transforms } from '_/ovirtapi'
+import { CREATE_DISK_FOR_VM, REMOVE_DISK, EDIT_VM_DISK } from '_/constants'
+
+import {
+  callExternalAction,
+  delay,
+  delayInMsSteps,
+  entityPermissionsToUserPermits,
+} from './utils'
 
 import {
   addDiskRemovalPendingTask,
   removeDiskRemovalPendingTask,
   extractErrorText,
   updateVmDisk,
-} from '../actions'
+  setVmDisks,
+} from '_/actions'
+
 import {
   setNewDiskDialogProgressIndicator,
   setNewDiskDialogErrorText,
   setNewDiskDialogDone,
-} from '../components/NewDiskDialog/actions'
+} from '_/components/NewDiskDialog/actions'
+
+import { canUserEditDisk } from '_/utils'
+
+function* transformAndPermitDiskAttachment (attachment) {
+  const internalDisk = Transforms.DiskAttachment.toInternal({ attachment, disk: attachment.disk })
+
+  internalDisk.userPermits = yield entityPermissionsToUserPermits(internalDisk)
+  internalDisk.canUserEditDisk = canUserEditDisk(internalDisk.userPermits)
+
+  return internalDisk
+}
+
+function* fetchVmDisks ({ vmId }) {
+  const diskattachments = yield callExternalAction('diskattachments', Api.diskattachments, { payload: { vmId } })
+
+  const internalDisks = []
+
+  if (diskattachments && diskattachments.disk_attachment) {
+    for (const attachment of diskattachments.disk_attachment) {
```

Comment:
can we use destructuring (or `.?` operator) to get the attachments? 
```
const { disk_attachment: attachments = []} = yield ...
// or
const attachements = diskattachments?.disk_attachment ?? [] 
```
can we use `yield all([...]) here ? i.e.
```
const internalDisks = yield all(attachments.map(a => call(transformAndPermitDiskAttachment, a))
```

Reply:
> can we use destructuring (or `.?` operator) to get the attachments?
> 
> ```
> const { disk_attachment: attachments = []} = yield ...
> // or
> const attachements = diskattachments?.disk_attachment ?? [] 
> ```

I don't feel the need to always destructure or pull out bits of objects.  The if condition using `?.` does make sense.

> 
> can we use `yield all([...]) here ? i.e.
> 
> ```
> const internalDisks = yield all(attachments.map(a => call(transformAndPermitDiskAttachment, a))
> ```

We could but it wouldn't matter since there are no async actions in `transformAndPermitDiskAttachment()`.  I'd really only use `yield all(...)` if there are async/fetch bits.